solr/views.py

```diff
@@ -42,9 +43,34 @@ def set_cookies(request):
         cookie_name = current_app.config.get('SOLR_SERVICE_FORWARD_COOKIE_NAME')
         cookie = {cookie_name: request.cookies.get(cookie_name, 'session')}
         return cookie if cookie[cookie_name] else None
-
+    
     @staticmethod
-    def cleanup_solr_request(payload):
+    def apply_protective_filters(payload, user_id, protected_fields):
+        """
+        Adds filters to the query that should limit results to conditions
+        that are associted with the user_id+protected_field. If a field is
+        not found in the db of limits, it will not be returned to the user
+        
+        :param payload: raw request payload
+        :param user_id: string, user id as known to ADS API
+        :param protected_fields: list of strings, fields
+        """
+        filters = set()
+        fl = payload.get('fl', 'id')
+        fq = payload.get('fq', [])
+        if not isinstance(fq, list):
+            fq = [fq]
+        payload['fq'] = fq
+        
+        for f in db.session.query(Limits).filter(Limits.uid==user_id, or_(Limits.field==x for x in protected_fields)).all():
+            if f.filter:
+                fl = u'{0},{1}'.format(fl, f.field)
+                fq.append(unicode(f.filter))
```

Comment:
Is there any protection on the initial `fq` given by the user? For example, could I put `fq: [u'bibstem:AA']` in `fq` to bypass the limits step here? I can achieve this in the tests, but wondered if I'm missing logic somewhere.

```bash
First list contains 1 additional elements.
First extra element 1:
bibstem:apj

- [u'bibstem:AA', u'bibstem:apj']
+ [u'bibstem:apj']
```